### PR TITLE
Endian fixes for the Tiberian Dawn coord getter and manipulation functions

### PR DESCRIPTION
**Pull request by isojalka**

### CODE REVIEW

File: tiberiandawn/function.h [hunk 1]
```diff
@@ -765,36 +765,68 @@ inline int Cell_Y(CELL cell)
...
+    coord2high = (coord2>>16)&0xffff;
+
+    return (COORDINATE)MAKE_LONG(coord1high + coord2high, coord1low + coord2low);
 }
 
 inline COORDINATE Coord_Sub(COORDINATE coord1, COORDINATE coord2)
 {
-    return (COORDINATE)MAKE_LONG((*((short*)(&coord1) + 1) - *((short*)(&coord2) + 1)),
-                                 (*((short*)(&coord1)) - *((short*)(&coord2))));
+    unsigned short coord1low;
```
Comment (tomsons26) [inline tiberiandawn/function.h]: def not
(COORDINATE)MAKE_LONG(Coord_X(coord1) - Coord_X(coord2), Coord_Y(coord1) - Coord_Y(coord2));

File: tiberiandawn/function.h [hunk 2]
```diff
@@ -765,36 +765,68 @@ inline int Cell_Y(CELL cell)
...
 }
 
 inline COORDINATE Coord_Snap(COORDINATE coord)
 {
-    return (COORDINATE)MAKE_LONG((((*(((unsigned short*)&coord) + 1)) & 0xFF00) | 0x80),
-                                 (((*((unsigned short*)&coord)) & 0xFF00) | 0x80));
+    unsigned short coordlow;
+    unsigned short coordhigh;
+
+    coordlow = (coord>>0)&0xffff;
```
Comment (tomsons26) [inline tiberiandawn/function.h]: (COORDINATE)MAKE_LONG(((Coord_X(coord) & 0xFF00) | 0x80), ((Coord_Y(coord) & 0xFF00) | 0x80));
i think..

File: tiberiandawn/function.h [hunk 3]
```diff
@@ -765,36 +765,68 @@ inline int Cell_Y(CELL cell)
...
+    unsigned short coord1high;
+    unsigned short coord2low;
+    unsigned short coord2high;
+
+    coord1low = (coord1>>0)&0xffff;
+    coord1high = (coord1>>16)&0xffff;
+    coord2low = (coord2>>0)&0xffff;
+    coord2high = (coord2>>16)&0xffff;
+
+    return (COORDINATE)MAKE_LONG((coord1high + coord2high + 1) >> 1, (coord1low + coord2low) >> 1);
```
Comment (tomsons26) [inline tiberiandawn/function.h]: (COORDINATE)MAKE_LONG((Coord_X(coord1) + Coord_X(coord2)) >> 1, (Coord_Y(coord1) + Coord_Y(coord2)) >> 1);

File: tiberiandawn/function.h [hunk 4]
```diff
@@ -765,36 +765,68 @@ inline int Cell_Y(CELL cell)
...
+    unsigned short coord1high;
+    unsigned short coord2low;
+    unsigned short coord2high;
+
+    coord1low = (coord1>>0)&0xffff;
+    coord1high = (coord1>>16)&0xffff;
+    coord2low = (coord2>>0)&0xffff;
+    coord2high = (coord2>>16)&0xffff;
+
+    return (COORDINATE)MAKE_LONG(coord1high + coord2high, coord1low + coord2low);
```
Comment (tomsons26) [inline tiberiandawn/function.h]: def not
(COORDINATE)MAKE_LONG(Coord_X(coord1) + Coord_X(coord2), Coord_Y(coord1) + Coord_Y(coord2));

**Comment by isojalka** [comment]
Amended the pull request based on your suggestions, thanks!

**Comment by OmniBlade** [comment]
@isojalka Is this PR still relevant since the merge of the RA code backport?

**Comment by isojalka** [comment]
No, this PR is indeed now obsolete, but PR883 fixes endian issues introduced by copying over the Red Alert code.

**Pull request closed without merging**